src/pages/api/answer.ts

```diff
@@ -1,23 +1,41 @@
 import {NextApiRequest, NextApiResponse} from 'next'
 import {convertkitAxios} from 'utils/axios-convertkit-api'
 import fetchConvertkitSubscriberFromServerCookie from 'utils/fetch-convertkit-subscriber'
+import isEmpty from 'lodash/isEmpty'
 
 const answer = async (req: NextApiRequest, res: NextApiResponse) => {
   if (req.method === 'POST') {
     try {
       console.log(req.body)
-      const {tagId} = req.body
+      const {tagId, quiz} = req.body
       const cookieHeader = req.headers.cookie as string
-      const [
-        subscriber,
-        ckCookie,
-      ] = await fetchConvertkitSubscriberFromServerCookie(cookieHeader)
+      const [subscriber, ckCookie] =
+        await fetchConvertkitSubscriberFromServerCookie(cookieHeader)
 
+      // Subscribe user to tagId
       await convertkitAxios.post(`/tags/${tagId}/subscribe`, {
         api_key: process.env.NEXT_PUBLIC_CONVERTKIT_TOKEN,
         email: subscriber.email_address,
       })
 
+      // Create Quiz answers field if it doesn't exist yet
+      if (!subscriber.fields.quiz_answers) {
+        await convertkitAxios.post(`/custom_fields`, {
+          api_secret: process.env.CONVERTKIT_API_SECRET,
+          label: 'Quiz answers',
+        })
+      }
```

Comment:
we can use `/custom_fields` endpoint to create a new custom field if it doesn't exist to avoid creating it manually.

Reply:
I'd want granular per quiz field in CK, not a blob of quiz answers

Reply:
the reasoning being that we will use these for survey results, segmenting, and personalization in convertkit, not just as a convenient place to dump data, if that makes sense.

Reply:
updated 👍